src/ir/template.rs

```diff
@@ -333,17 +332,11 @@ impl IsOpaque for TemplateInstantiation {
     }
 }
 
-impl<'a> CanDeriveCopy<'a> for TemplateInstantiation {
-    type Extra = ();
-
-    fn can_derive_copy(&self, ctx: &BindgenContext, _: ()) -> bool {
-        self.definition.can_derive_copy(ctx, ()) &&
-        self.args.iter().all(|arg| arg.can_derive_copy(ctx, ()))
-    }
-
-    fn can_derive_copy_in_array(&self, ctx: &BindgenContext, _: ()) -> bool {
-        self.definition.can_derive_copy_in_array(ctx, ()) &&
-        self.args.iter().all(|arg| arg.can_derive_copy_in_array(ctx, ()))
+impl HasArray for TemplateInstantiation {
+    /// Is this has array?
+    fn has_array(&self, ctx: &BindgenContext) -> bool {
+        let resolved_item = ctx.resolve_item(self.template_definition());
+        resolved_item.as_type().map_or(false, |ty| ty.has_array(ctx))
```

Comment:
Do we need to consider the template arguments here as well? The old `can_derive_copy_in_array` for `TemplateInstantiation` also checked that each of the args could derive copy in an array, but I'm not sure if that translates to `HasArray` or not.